cpp/ql/test/library-tests/dataflow/taint-tests/test_default_diff.ql

```diff
@@ -0,0 +1,95 @@
+import cpp
+import semmle.code.cpp.dataflow.TaintTracking
+import semmle.code.cpp.security.Security
+import semmle.code.cpp.ir.dataflow.DefaultTaintTracking as IRDefaultTaintTracking
+import semmle.code.cpp.dataflow.DataFlow as ASTDataFlow
+import semmle.code.cpp.ir.dataflow.DataFlow as IRDataFlow
+
+// Only perform taint tracking from parameters, global variables or field variables
+private class TestSecurityOptions extends SecurityOptions {
+  override predicate isUserInput(Expr expr, string cause) {
+    (
+      exists(Parameter p | p.getAnAccess() = expr) or
+      expr.(VariableAccess).getTarget().getParentScope() instanceof GlobalNamespace or
+      exists(Field f | f.getAnAccess() = expr)
```

Comment:
Why not use the default sources only? Adding all these sources will certainly show us more bugs we can fix, but I imagine there are already many bugs visible if we use the default sources only, and those will be the most important bugs.

Reply:
This was an attempt to reduce the number of taint sources (i.e., by not considering local variables as sources of taint). But seeing the output after just using the default sources I see that it had the opposite effect.